tracing/src/main/java/com/palantir/tracing/TraceState.java

```diff
@@ -50,6 +54,7 @@ private TraceState(String traceId, @Nullable String requestId, @Nullable String
         this.traceId = traceId;
         this.requestId = requestId;
         this.forUserAgent = forUserAgent;
+        this.traceLocals = new ConcurrentHashMap<>();
```

Comment:
How much does this allocation add to perf overhead of tracing? Should we lazily initialize this map only when trace locals are used?

Reply:
:+1: we should use an atomic-reference-field-updater to thread-safely initialize `traceLocals` to a new map as needed

Reply:
done